src/server/api/controllers/candidates.controller.js

```diff
@@ -43,8 +43,26 @@ const createCandidate = async (userId, boardId, newCandidate) => {
   return createNewCandidate;
 };
 
+const getCandidates = async (userId, boardId) => {
+  if (!Number.isInteger(Number(userId)) || !Number.isInteger(Number(boardId))) {
+    throw new InvalidIdError('Id should be an integer');
+  }
+  const candidates = await knex('candidates')
+    .join('boards', 'candidates.boardId', '=', 'boards.id')
+    .join('ballots', 'ballots.candidateId', '=', 'candidates.id')
+    .select('candidates.name', 'ballots.rank')
```

Comment:
Please also select the `candidateId` it will be needed for further communication.